runtime/parser/declaration.go

```diff
@@ -958,6 +969,106 @@ func parseCompositeOrInterfaceDeclaration(
 	}
 }
 
+func parseAttachmentDeclaration(
+	p *parser,
+	access ast.Access,
+	accessPos *ast.Position,
+	docString string,
+) (ast.Declaration, error) {
+	startPos := p.current.StartPos
+	if accessPos != nil {
+		startPos = *accessPos
+	}
+
+	// Skip the attachment keyword
+	p.next()
+
+	p.skipSpaceAndComments()
+	if !p.current.Is(lexer.TokenIdentifier) {
+		return nil, p.syntaxError(
+			"expected %s, got %s",
+			lexer.TokenIdentifier,
+			p.current.Type,
+		)
+	}
+	identifier, err := p.mustIdentifier()
```

Comment:
`p.mustIdentifier()` also checks for `p.current.Is(lexer.TokenIdentifier)`. So the above check is redundant.